src/NUnitFramework/tests/Assertions/AssertThrowsAsyncTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Threading.Tasks;
+using NUnit.TestUtilities;
+
+namespace NUnit.Framework.Tests.Assertions
+{
+    [TestFixture]
+    public class AssertThrowsAsyncTests
+    {
```

Comment:
> Note that our general practice is to include framework files in all the framework builds but #if them out where not wanted. That facilitates searching the projects for all the #if statements.

Regarding this, wouldn't it be simpler to use a shared project (.shproj)? Basically you have a dummy (not buildable) project that just acts as a container for source files, and you reference it from all the projects that need those files. It achieves the same result as including the same source files in several projects, but is easier to work with, because you only need to add new files to the shared project, instead of manually adding them to each project that needs them.